### DISCUSSION TITLE
Autocomplete only working on a line by line basis

### DISCUSSION THREAD

**Discussion by cgat** [community/community#7032] [Copilot Conversations] [answered]
In the example I've seen, copilot will suggest a full function body after entering a function signature, but I'm only getting one line at a time.

For examples
```js
function calculateDaysBetweenDates(begin, end) {
```

Does not give me an autocomplete suggestion.

If I press enter and go to the next line I'll see a one line suggestion

![image](https://user-images.githubusercontent.com/3103805/139511165-89b39715-8a5e-4b6a-b475-69c3637b5806.png)

Tabbing will complete this line.

If I hover over the suggestion, the copilot dialogue opens up. Next and Prev are disabled. If I click "Open Github Copilot", a split view pane appears with a bunch of fully function body suggestions.

**Comment by MileyHollenberg** [comment, marked as answer]
For me it sometimes fills the full body but more often than not it does it line by line. If you want to force a full completion you need to press Alt+Enter which will open up a new tab where you get up to 10 suggestions you can accept

**Comment by tiagosalema** [comment]
I couldn't have the `calculateDaysBetweenDates` function be suggested entirely either, but only line by line. However, I got a suggestion for the function `calculateDays`:

![image](https://i.imgur.com/S5xACqF.gif)

**Comment by cgat** [comment]
Thanks everyone. Yes, after a bit of use, I seem to be getting full body suggestions sometimes. A bit hit and miss, but it works.

**Comment by MulleDK19** [comment]
It completes entire blocks unless you've typed something inside. Remove the var and it'll complete the whole function.

**Comment by MarvinXu** [comment]
I found it can complete the whole function in a vue SFC, but not in a ts file. Is there any config that can control this behavior?

![image](https://user-images.githubusercontent.com/7382357/161715681-db53ed86-ac7d-422f-8b69-48f3d52a2191.png)
![image](https://user-images.githubusercontent.com/7382357/161715323-ae4170ff-e485-4ff6-82f0-7951e151b04b.png)

**Comment by chirag127** [comment]
What's the current state of the problem?
